### PR TITLE
WT-11116 Correct layout of __wt_page

### PR DESCRIPTION
**Pull request by marcbutler**
Corrects layout of __wt_page. Thereby reducing unnecessary padding and bringing the modify field back within the first 64 bytes of struct consistent with code comments.

### CODE REVIEW

File: src/include/btmem.h [hunk 1]
```diff
@@ -712,7 +700,17 @@ struct __wt_page {
...
+    ((page)->type == WT_PAGE_COL_INT || (page)->type == WT_PAGE_ROW_INT)
+#define WT_PAGE_INVALID 0       /* Invalid page */
+#define WT_PAGE_BLOCK_MANAGER 1 /* Block-manager page */
+#define WT_PAGE_COL_FIX 2       /* Col-store fixed-len leaf */
+#define WT_PAGE_COL_INT 3       /* Col-store internal page */
+#define WT_PAGE_COL_VAR 4       /* Col-store var-length leaf page */
+#define WT_PAGE_OVFL 5          /* Overflow page */
+#define WT_PAGE_ROW_INT 6       /* Row-store internal page */
+#define WT_PAGE_ROW_LEAF 7      /* Row-store leaf page */
+    uint8_t type;               /* Page type */
```
Comment (keitharnoldsmith) [inline src/include/btmem.h:714]: I would keep the `uint8_t unused;` as a reminder to future readers that there is another byte of padding in here. It also matches the way we annotate such padding in other places.

**Review by keitharnoldsmith — APPROVED** [review]
One change request. I'm approving since it would LGTM with that, and there's nothing tricky to re-check.

File: src/include/btmem.h (hunk 1 above)
Comment (marcbutler) [reply src/include/btmem.h:714]: Thank you both for the approval. Would a comment suffice instead of an explicit padding field?
Note: I'm just waiting for OOB discussions to conclude before I make the change.

cc @sueloverso

Comment (sueloverso) [reply src/include/btmem.h:714]: I'm good with whatever the majority decided on slack. Both have pros/cons and work for me.

Comment (keitharnoldsmith) [reply src/include/btmem.h:714]: I also don't have strong feelings on this point. But I do think it's good to be consistent across the code. I would suggest including the explicit `unused` field in this PR for consistency.

Then you can make the case for comments on compiler padding, or some other alternative, and create a new ticket/PR to convert to the resulting standard.

**Comment by keitharnoldsmith** [comment]
@marcbutler pinged me to ask why I wanted to include explicit padding, given that the problem in this bug is that we changed the structure and the padding field caused the part we wanted to fit on a single cache line to grow past that. My response turned out to be long enough that I figured it would be better to capture here on the PR for posterity and/or wider exposure.

tl;dr On further review, I'm OK just leaving a comment to indicate the space rather than making it explicit.

In general, I think we should do similar things in similar ways in the code. I'm aware of other places where we care about structure size and/or alignment and we include explicit padding. So I was basically arguing that explicit padding here would keep the style consistent.

But I just went though and looked more carefully and where and how we use explicit padding, and I think I was over generalizing. Searching for structures that include fields named `unused` or `padding` (a point we're not consistent about :-), I found that our existing usage falls into a couple cases that don't match what you're addressing here:

- Persistent structures such as block and page headers, log records, etc. These need to have fixed structure. We can't have a change of compiler (or compiler version) result in incompatible on disk structures.
- Places we have padding adjacent to flag words. In these cases the padding is essentially calling out "here is some extra space if we need additional flag values"
- A bit of [padding](https://github.com/wiredtiger/wiredtiger/blob/5c95825b6ccd29e05ee1e985a32038aff9a09acd/src/include/mutex.h#L103) in `__wt_spinlock` that seems unnecessary.

So this seems like a unique situation -- a case where we want a set of fields to land in the same cache line and the fields don't themselves account for the entire cache line, leaving the question of how to represent the required padding. Since there isn't a style consistency question, I'm happy to go with your suggestion that we keep the padding implicit.

Feel free to merge the PR as it stands.
